app/main/views/login.py

```diff
@@ -62,6 +62,10 @@ def process_login():
         login_user(user)
         current_app.logger.info('login.success: {user}', extra={'user': user_logging_string(user)})
         check_terms_acceptance()
+        if current_user.role == 'buyer':
+            user = User.load_user(data_api_client, current_user.id)
```

Comment:
```suggestion
            user = User.load_user(data_api_client, current_user.id)
```
the user variable is already used... does the original user variable have everything you need?

Reply:
It does not - when calling `login_user()`, the user loader is only called on next page request (during Flask init) so this is needed to get the extra properties right after logging in.